### PR TITLE
Add Nginx Support

### LINKED ISSUE #505: Feature request: nginx.conf
I :heart: this package, and would :heart: to be able to beautify nginx.conf files with it

### PR DESCRIPTION
**Pull request by denysvitali**
### Nginx support
This PR adds nginx.conf support, it uses my [nginxbeautify](https://github.com/denysvitali/nginxbeautify) module, based on [nginxbeautifier](https://github.com/vasilevich/nginxbeautifier) by @vasilevich

### Closes
#505 

### Checklist

Check all those that are applicable and complete.

- [X] Merged with latest `master` branch
- [X] Added examples for testing to [examples/ directory](examples/)
- [ ] Travis CI passes (Mac support)
- [ ] AppVeyor passes (Windows support)

### CODE REVIEW

**Comment by mccrodp** [comment]
This would be a great feature, looking for something similar at the moment, was surprised atom-beautify didn't already perform this. Thanks for taking the steps towards this!

**Comment by denysvitali** [comment]
@mccrodp It works, it just have to be merged. If you want you can try it by using the one from [my repo](https://github.com/denysvitali/atom-beautify/) 👍

**Comment by mccrodp** [comment]
Thanks @denysvitali. Might be good to move the remaining "Travis CI passes (Mac support)"
and "AppVeyor passes (Windows support)" out to separate follow up issues and get this one merged! 👍

File: src/languages/nginx.coffee [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  Supported extensions
+  ###
+  extensions: [
+    "conf"
+  ]
+
+  defaultBeautifier: "Nginx Beautify"
+
+  options:
+    spaces:
```
Comment (Glavin001) [inline src/languages/nginx.coffee]: I try to keep the same names for the same options, such as `indent_size` and `indent_char`: https://github.com/Glavin001/atom-beautify/blob/master/src/languages/javascript.coffee#L35-L43

Then in the beautifier itself apply some transformations to the option values, like https://github.com/Glavin001/atom-beautify/blob/master/src/beautifiers/prettydiff.coffee#L10-L44

Comment (Glavin001) [inline src/languages/nginx.coffee]: Please remove `spaces` and `tabs` from the language, use `indent_size` and `indent_char` (copy from `JavaScript` language), and then change in the beautifier.

File: src/beautifiers/nginx-beautify.coffee [hunk 2]
```diff
@@ -0,0 +1,22 @@
+"use strict"
+Beautifier = require('./beautifier')
+
+module.exports = class NginxBeautify extends Beautifier
+  name: "Nginx Beautify"
+  link: "https://github.com/denysvitali/nginxbeautify"
+
+  options: {
+    Nginx: true
```
Comment (Glavin001) [inline src/beautifiers/nginx-beautify.coffee]: Apply transformations to `indent_size` and `indent_char` here. See https://github.com/Glavin001/atom-beautify/blob/master/src/beautifiers/prettydiff.coffee#L10-L44 for examples.

Try something like:

```
spaces: ["indent_size", "indent_char", (indent_size, indent_char) -> if indent_char is "\t" then 0 else indent_size
tabs: ["indent_size", "indent_char", (indent_size, indent_char) -> if indent_char is "\t" then indent_size else 0
```

**Comment by Glavin001** [comment]
I ended up making the changes myself.

Thanks for contributing!

**Comment by Glavin001** [comment]
Published to v0.29.19

**Comment by denysvitali** [comment]
Sorry about that, I was quite busy the last week 😢

**Comment by Glavin001** [comment]
No worries, I've been missing in action for months because of being busy with work and school, so I absolutely understand 😄.  Let me know if there are any problems with the recent release!

**Comment by daggerjames** [comment]
Since nginxbeautify is a node.js package, why not pre-install it?

The beautify works by requiring the package, rather than through CLI. Thus, installing the package globally seems unnecessary, especially when you installed multiple versions of node.js

**Comment by Glavin001** [comment]
@daggerjames the `nginxbeautify` should be preinstalled, is it not? 
See https://github.com/Glavin001/atom-beautify/pull/1494/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R160

**Comment by Glavin001** [comment]
Fixed the documentation, good eye @daggerjames. Thanks!

**Pull request merged**